php/ext/google/protobuf/upb.c

```diff
@@ -7153,7 +7154,8 @@ size_t run_decoder_vm(upb_pbdecoder *d, const mgroup *group,
         CHECK_SUSPEND(upb_sink_startsubmsg(outer->sink, arg, &d->top->sink));
       )
       VMCASE(OP_ENDSUBMSG,
-        CHECK_SUSPEND(upb_sink_endsubmsg(d->top->sink, arg));
+        upb_sink subsink = (d->top + 1)->sink;
```

Comment:
Is this `+ 1` safe? I am not sure about the implication here.

Reply:
I think it's safe. It's moving pointer from parent's sink to child's sink.
Because, it's handling sub message here, the sub sink should have already be created.